lib/elixir/lib/exception.ex

```diff
@@ -1091,18 +1091,85 @@ defmodule Protocol.UndefinedError do
 end
 
 defmodule KeyError do
-  defexception [:key, :term]
+  defexception [:key, :term, :message]
 
   @impl true
-  def message(exception) do
-    msg = "key #{inspect(exception.key)} not found"
+  def message(exception = %{message: nil}), do: message(exception.key, exception.term)
+  def message(%{message: message}), do: message
+
+  def message(key, term) do
+    msg = "key #{inspect(key)} not found"
```

Comment:
Minor nitpick: what do you think about using `message` instead of `msg` (like the function above) and `distance` instead of `dist` in the functions below?

Reply:
Yeah, works for me!